python/cugraph/tests/test_renumber.py

```diff
@@ -186,33 +218,135 @@ def test_renumber_files_col(graph_file):
+    gdf["src_old"] = sources
+    gdf["dst_old"] = destinations
+    gdf["src"] = sources + translate
+    gdf["dst"] = destinations + translate
+
+    numbering = NumberMap()
+    numbering.from_dataframe(gdf, ["src", "src_old"], ["dst", "dst_old"])
+
+    renumbered_df = numbering.add_internal_vertex_id(
+        numbering.add_internal_vertex_id(
+            gdf, "src_id", ["src", "src_old"]
+        ),
+        "dst_id",
+        ["dst", "dst_old"],
+    )
+
+    check_src = numbering.from_internal_vertex_id(renumbered_df, "src_id")
+    check_dst = numbering.from_internal_vertex_id(renumbered_df, "dst_id")
+
+    assert check_src["src"].equals(check_src["0"])
+    assert check_src["src_old"].equals(check_src["1"])
+    assert check_dst["dst"].equals(check_dst["0"])
+    assert check_dst["dst_old"].equals(check_dst["1"])
+
+
+@pytest.fixture()
+def client_setup():
+    cluster = LocalCUDACluster()
+    client = Client(cluster)
+    client
+
+
+# Test all combinations of default/managed and pooled/non-pooled allocation
+@pytest.mark.parametrize("graph_file", utils.DATASETS)
+def test_opg_renumber(graph_file, client_setup):
+    gc.collect()
+
+    M = utils.read_csv_for_nx(graph_file)
+    sources = cudf.Series(M["0"])
+    destinations = cudf.Series(M["1"])
 
     translate = 1000
 
     gdf = cudf.DataFrame()
-    gdf['src_old'] = sources
-    gdf['dst_old'] = destinations
-    gdf['src'] = sources + translate
-    gdf['dst'] = destinations + translate
+    gdf["src_old"] = sources
+    gdf["dst_old"] = destinations
+    gdf["src"] = sources + translate
+    gdf["dst"] = destinations + translate
+
+    ddf = dask.dataframe.from_pandas(gdf, npartitions=2)
+
+    numbering = NumberMap()
+    numbering.from_dataframe(ddf, ["src", "src_old"], ["dst", "dst_old"])
+    renumbered_df = numbering.add_internal_vertex_id(
+        numbering.add_internal_vertex_id(ddf, "src_id", ["src", "src_old"]),
+        "dst_id",
+        ["dst", "dst_old"],
+    )
+
+    check_src = numbering.from_internal_vertex_id(
+        renumbered_df, "src_id"
+    ).compute()
+    check_dst = numbering.from_internal_vertex_id(
+        renumbered_df, "dst_id"
+    ).compute()
+
+    assert check_src["0"].to_pandas().equals(check_src["src"].to_pandas())
+    assert check_src["1"].to_pandas().equals(check_src["src_old"].to_pandas())
+    assert check_dst["0"].to_pandas().equals(check_dst["dst"].to_pandas())
+    assert check_dst["1"].to_pandas().equals(check_dst["dst_old"].to_pandas())
 
-    src, dst, numbering = cugraph.renumber_from_cudf(
-        gdf, ['src', 'src_old'], ['dst', 'dst_old'])
 
-    for i in range(len(gdf)):
-        assert sources[i] == (numbering['0'].iloc[src[i]] - translate)
-        assert destinations[i] == (numbering['0'].iloc[dst[i]] - translate)
+# Test all combinations of default/managed and pooled/non-pooled allocation
+@pytest.mark.parametrize("graph_file", utils.DATASETS)
+def test_opg_renumber2(graph_file, client_setup):
```

Comment:
Can we add an MG pipeline test involving cuGraph `Graph` and `from_dask_cudf_edgelist`?

Reply:
I have one that's not checked in, I can add it to the PR.